addons/l10n_tr_nilvera_edispatch/static/src/views/list_view/ereceipt_upload.js

```diff
@@ -0,0 +1,114 @@
+    static components = { FileUploader };
+
+    setup() {
+        this.orm = useService("orm");
+        this.notification = useService("notification");
+        this.action = useService("action");
+        this.attachmentIdsToProcess = [];
+        this.invalidAttachments = [];
+    }
+
+    async onEreceiptFileUpload(file) {
+        if (file.type != "text/xml") {
+            this.invalidAttachments.push(file.name);
+            return;
+        }
+        const file_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        };
+        const [attachmentId] = await this.orm.create("ir.attachment", [file_data]);
+        this.attachmentIdsToProcess.push(attachmentId);
+    }
+
+    async onEreceiptUploadComplete() {
+        if (this.invalidAttachments.length !== 0) {
+            this.notification.add(
+                `The file(s): ${this.invalidAttachments.join(", ")} must be of type XML.`,
+                {
+                    title: "Only XML files can be uploaded",
+                    type: "danger",
+                }
+            );
+            this.invalidAttachments = [];
+        }
+        if (this.attachmentIdsToProcess.length === 0) {
+            return;
+        }
+        try {
+            const result = await this.orm.call("stock.picking", "l10n_tr_import_ereceipts", [
+                "",
+                this.attachmentIdsToProcess,
+            ]);
+
+            if (result) {
+                if (result.action) {
+                    this.notification.add("e-Receipt(s) Imported Successfully", {
+                        type: "success",
+                    });
+                    this.action.doAction(result.action);
+                }
+                if (result.skipped_xmls) {
+                    this.notification.add(
+                        `Error occured in reading following XML file(s): ${result.skipped_xmls.join(", ")}.`,
+                        { type: "danger", title: "e-Receipt(s) were not imported", sticky: true }
+                    );
+                }
+            }
+        } catch (e) {
+            this.notification.add(e.data.message, {
+                title: "Something went wrong. Please try again.",
+                type: "danger",
+            });
+        } finally {
+            this.attachmentIdsToProcess = [];
+        }
+    }
+}
+
+export class L10nTrNilveraEdispatchListController extends ListController {
+    setup() {
+        super.setup();
+        onWillStart(async () => {
+            const currentCompanyId = this.env.services.company.currentCompany.id;
+            this.data = await this.orm.searchRead(
+                "res.company",
+                [["id", "=", currentCompanyId]],
+                ["country_code"]
+            );
+            this.countryCode = this.data[0].country_code;
+        });
+    }
+
+    get isActionDisplayed() {
+        return (
+            this.countryCode === "TR" &&
```

Comment:
I find the context for restricted_picking_type_code a bit weird, but don't know how I would do it differently.
But if you're doing that, can't you do the same for the countryCode? (get the country_code of the env.company from the python).
It will avoid you doing the searchRead on company